### PR TITLE
feat:add mock package, improved test coverage

### PR DESCRIPTION
**Pull request by medgar-nitric**

### CODE REVIEW

File: src/main/java/io/nitric/api/document/Query.java [hunk 1]
```diff
@@ -350,9 +354,9 @@
...
      */
     public Query<T> where(String operand, String operator, Boolean value) {
-        Contracts.requireNonBlank(operand, "operand");
-        Contracts.requireNonBlank(operator, "operator");
-        Contracts.requireNonNull(value, "value");
+        Contracts.requireNonBlank(operand, OPERAND);
+        Contracts.requireNonBlank(operator, OPERATOR);
+        Contracts.requireNonNull(value, VALUE);
 
         expressions.add(new Expression(operand, operator, value));
```
Comment (tjholm) [inline src/main/java/io/nitric/api/document/Query.java:361]: Would we be able to de-duplicate these methods with a private unified method add new expressions, e.g.:

```java
private Query<T> whereObj(String operand, String operator, Object value) {
  Contracts.requireNonBlank(operand, OPERAND);
  Contracts.requireNonBlank(operator, OPERATOR);
  Contracts.requireNonNull(value, VALUE);

  expressions.add(new Expression(operand, operator, value));
  
  return this;
}

public Query<T> where(String operand, String operator, Bool value) {
  return this.whereObj(operand, operator, value);
}
```

File: src/main/java/io/nitric/faas/Faas.java [hunk 2]
```diff
@@ -82,7 +82,7 @@
     public void startFunction(NitricFunction function) {
         Contracts.requireNonNull(function, "function");
 
-        // FIXME: Uncoverable code without mocking static methods (need to include Powermock)
+        // FIXME: Unrecoverable code without mocking static methods (need to include Powermock)
```
Comment (tjholm) [inline src/main/java/io/nitric/faas/Faas.java:85]: This was a note on our inability to cover this code without use of powermock, probably just a spelling correction but 'Unrecoverable' is not correct.

**Review by tjholm — CHANGES_REQUESTED** [review]

File: src/main/java/io/nitric/faas/Faas.java (hunk 2 above)
Comment (medgar-nitric) [reply src/main/java/io/nitric/faas/Faas.java:85]: Opps thanks

File: src/main/java/io/nitric/api/document/Query.java (hunk 1 above)
Comment (medgar-nitric) [reply src/main/java/io/nitric/api/document/Query.java:361]: Yep, will give that a go.

**Review by tjholm — APPROVED** [review]

**Pull request merged**